frontend/src/components/maps/leaflet/Map.tsx

```diff
@@ -177,22 +180,28 @@ const Map: React.FC<MapProps> = ({
     });
   }, []);
 
-  // we need to namespace the keys as IDs are not enough here.
-  // the same ID could be found on both the parcel collection and building collection
-  const generateKey = (p: IProperty) => `${p.propertyTypeId === 0 ? 'parcel' : 'building'}-${p.id}`;
+  // load and prepare data
+  const points = createPoints(properties);
 
-  const renderMarker = (p: IProperty) => {
-    const icon = p.propertyTypeId === 0 ? greenIcon : blueIcon;
-    return (
-      <Marker
-        key={generateKey(p)}
-        position={[p.latitude, p.longitude]}
-        onClick={(e: any) => onMarkerClick?.(p)}
-        icon={icon}
-      />
-    );
+  // get map bounds
+  const updateMap = () => {
+    if (!mapRef?.current) {
+      return;
+    }
+    const b = mapRef.current.leafletElement.getBounds();
+    setBounds([
+      b.getSouthWest().lng,
+      b.getSouthWest().lat,
+      b.getNorthEast().lng,
+      b.getNorthEast().lat,
+    ]);
+    setZoom(mapRef.current.leafletElement.getZoom());
   };
 
+  useEffect(() => {
+    updateMap();
+  }, []);
+
```

Comment:
This effect keeps track of the map bounds and zoom level because clustering depends on them